### ISSUE TITLE
Unused assignment analysis isn't handling `??=` correctly

### ISSUE THREAD

**Issue by 333fred** [dotnet/roslyn#34414]
```C#
using System;

public class C
{
    public void M(object o = null /* Flagged as unused value */)
    {
        o ??= "Hello world";
        Console.WriteLine(o);
    }

    public void M2(object o)
    {
        o = GetObject(); // Flagged as unused value
        o ??= "Hello world";
        Console.WriteLine(o);
    }

    public void M3()
    {
        object o = GetObject(); // Not flagged as unused value?
        o ??= "Hello world";
        Console.WriteLine(o);
    }

    object GetObject() => throw null;
}
```

The first two are, for some reason, flagged as unused, but not the third. /cc @genlu @mavasani.

**Comment by sharwell** [comment]
Related to #34448

**Comment by mavasani** [comment]
This was fixed a while ago.
